app/src/docs/examples/fileUpload/FileUpload.example.tsx

```diff
@@ -1,54 +1,61 @@
 import * as React from 'react';
 import { DropSpot, FileCard } from '@epam/promo';
-import { FileUploadResponse, useUuiContext } from '@epam/uui';
+import { useUuiContext, FileUploadResponse } from '@epam/uui';
 import * as css from './FileUpload.scss';
 
-type AttachmentType = FileUploadResponse & { progress?: number };
-
 const ORIGIN = process.env.REACT_APP_PUBLIC_URL || '';
 
+interface NewFile extends Partial<File & FileUploadResponse> {
+    progress?: number;
+    abortXHR?: () => void;
+}
+
+let tempIdCount = 0;
+
 export default function FileUploadExample() {
     const { uuiApi } = useUuiContext();
-    const [attachments, setAttachments] = React.useState<AttachmentType[]>([]);
+    const [attachments, setAttachments] = React.useState<NewFile[]>([]);
 
     const trackProgress = (progress: number, id: number) => {
-        const file = attachments.find(i => i.id === id);
-        setAttachments(attachments.map(f => f.id === file.id ? { ...file, progress } : f));
+        setAttachments((attachments) => attachments.map(item => item.id === id ? { ...item, progress } : item));
     };
 
-    const updateFile = (file: AttachmentType, id: number) => {
-        setAttachments(attachments.map(f => f.id === id ? file : f));
+    const updateFile = (file: NewFile, id: number) => {
+        setAttachments((attachments) => attachments.map((item) => item.id === id ? file : item));
     };
 
-    const deleteFile = (index: number) => {
-        setAttachments(attachments.filter((_, i) => i !== index));
+    const deleteFile = (file: NewFile) => {
+        setAttachments((attachments) => attachments.filter((item) => item.id !== file.id));
     };
 
     const uploadFile = (files: File[]) => {
-        const newAttachments = attachments.slice();
+        const newAttachments = [...attachments];
+
+        files.map((file: File) => {
+            const tempId = tempIdCount - 1;
+            tempIdCount -= 1;
+            const newFile: NewFile = { id: tempId, name: file.name, progress: 0, size: file.size, uploadError: { isError: false } };
+            newAttachments.push(newFile);
 
-        files.map((file, index) => {
-            const tempId = index - 1;
-            attachments.push({ id: tempId, name: file.name, size: file.size });
             uuiApi.uploadFile(ORIGIN.concat('/uploadFileMock'), file, {
                 onProgress: progress => trackProgress(progress, tempId),
-                getXHR: xhr => xhr.setRequestHeader('my-header', 'value'),
-            }).then(res => updateFile({ ...res, progress: 100 }, tempId));
+                getXHR: (xhr) => {
+                    newFile.abortXHR = () => xhr.abort();
+                },
+            })
+                .then((res) => updateFile({ ...res, progress: 100, uploadError: { isError: false } }, tempId))
+                .catch(() => updateFile({ ...newFile, progress: 100, uploadError: { isError: true, message: 'Upload error' } }, tempId));
```

Comment:
we don't need to write here error info, it will be passed from ...newFile. And the same for .then